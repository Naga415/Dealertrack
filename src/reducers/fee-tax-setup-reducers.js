import extend from "lodash/extend";
import { FETCH_FEE_TAX_SETUP_DATA_SUCCESS } from "../actions/taxes-page-actions";

export const initialState = {
  data: []
};
const feeTaxSetup = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_FEE_TAX_SETUP_DATA_SUCCESS:
      return extend({}, state, {
        data: action.payload.data
      });
    default:
      return state;
  }
};

export default feeTaxSetup;
