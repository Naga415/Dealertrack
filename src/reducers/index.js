import {combineReducers} from 'redux';
import taxesPage from './taxes-page-reducers';

const rootReducer = combineReducers({
  taxesPage
});

export default rootReducer;
