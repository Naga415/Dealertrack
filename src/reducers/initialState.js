// This is the app's initial state. Note that reducers each handle a slice of this state.
// Composing all initial state here gives us a clear spot of reference that displays the shape
// of our entire store.
import Deal from '../models/Deal';
import Customer from '../models/Customer';

export default {
  customer: new Customer(),// see Customer model for definition
  deal: new Deal(),// see Deal model for definition
  loading: true,
  ratesAndResiduals: {
    lenders: [],
    selectedLender: null,
    programs: [],
    selectedProgram: null, // see Program model for definition
    incentives: [],
    selectedIncentives: [] // see Incentive model for definition
  }
};
