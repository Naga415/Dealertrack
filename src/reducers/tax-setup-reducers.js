import extend from 'lodash/extend';
import { UPDATE_TAX_SETUP_INPUT, FETCH_TAX_SETUP_DATA_SUCCESS } from '../actions/taxes-page-actions';
import { convertNullOrUndefinedToEmptyStrings } from '../utils/reducer-utils';

export const initialState = {
  retailFlat: "0",
  retailTradeCredit: null,
  retailTradeCreditAmount: "0",
  retailTaxPercent: "0",
  retailTaxMethod: null,
  retailTaxFees: false,
  retailTaxProducts: false,
  capitalize: false,
  leaseFlat: "0",
  leaseTradeCredit: null,
  leaseTradeCreditAmount: "0",
  leaseTaxPercent: "0",
  leaseTaxMethod: null,
  leaseTaxFees: false,
  leaseTaxProducts: false,
  monthlyUseTaxPercent: "0",
  monthlyTaxMethod: null,
  monthlyTaxFees: false,
  monthlyTaxProducts: false,
  capReductionTaxPercent: "0",
  capReductionTaxMethod: null,
  capTaxFees: false,
  capTaxProducts: false
};

const taxSetup = ( state = initialState, action) => {
  switch (action.type){
    case FETCH_TAX_SETUP_DATA_SUCCESS: 
      let retailTaxFields = action.payload.data.find(e => e.tax_type_name === 'retail');
      let monthlyUseTaxFields = action.payload.data.find(e => e.tax_type_name === 'mut');
      let capReductionTaxFields = action.payload.data.find(e => e.tax_type_name === 'crt');
      let leaseTaxFields = action.payload.data.find(e => e.tax_type_name === 'lease');

      retailTaxFields = convertNullOrUndefinedToEmptyStrings(retailTaxFields, [
        'flat_amt',
        'tax_rate',
        'trade_credit_amount'
      ]);

      leaseTaxFields = convertNullOrUndefinedToEmptyStrings(leaseTaxFields, [
        'flat_amt',
        'tax_rate',
        'trade_credit_amount'
      ]);

      monthlyUseTaxFields = convertNullOrUndefinedToEmptyStrings(monthlyUseTaxFields, [
        'tax_rate',
      ]);

      capReductionTaxFields = convertNullOrUndefinedToEmptyStrings(capReductionTaxFields, [
        'tax_rate'
      ]);

      return extend({}, state, {
        retailFlat: retailTaxFields['flat_amt'],
        retailTradeCredit: retailTaxFields['trade_credit_id'] && { value: retailTaxFields['trade_credit_id'], label: retailTaxFields['trade_credit_name']},
        retailTradeCreditAmount: retailTaxFields['trade_credit_amount'],
        retailTaxPercent: retailTaxFields['tax_rate'],
        retailTaxMethod: retailTaxFields['method_id'] && { value: retailTaxFields['method_id'], label: retailTaxFields['method_name']},
        retailTaxFees: (retailTaxFields['is_fee'] === 'Y'),
        retailTaxProducts: (retailTaxFields['is_product'] === 'Y'),
        leaseFlat: leaseTaxFields['flat_amt'],
        leaseTradeCredit: leaseTaxFields['trade_credit_id'] && { value: leaseTaxFields['trade_credit_id'], label: leaseTaxFields['trade_credit_name']},
        leaseTradeCreditAmount: leaseTaxFields['trade_credit_amount'],
        leaseTaxPercent: leaseTaxFields['tax_rate'],
        leaseTaxMethod: leaseTaxFields['method_id'] && { value: leaseTaxFields['method_id'], label: leaseTaxFields['method_name']},
        leaseTaxFees: (leaseTaxFields['is_fee'] === 'Y'),
        leaseTaxProducts: (leaseTaxFields['is_product'] === 'Y'),
        monthlyUseTaxPercent: monthlyUseTaxFields['tax_rate'],
        monthlyTaxMethod: monthlyUseTaxFields['method_id'] && { value: monthlyUseTaxFields['method_id'], label: monthlyUseTaxFields['method_name']},
        monthlyTaxFees: (monthlyUseTaxFields['is_fee'] === 'Y'),
        monthlyTaxProducts: (monthlyUseTaxFields['is_product'] === 'Y'),
        capReductionTaxPercent: capReductionTaxFields['tax_rate'],
        capReductionTaxMethod: capReductionTaxFields['method_id'] && { value: capReductionTaxFields['method_id'], label: capReductionTaxFields['method_name']},
        capTaxFees: (capReductionTaxFields['is_fee'] === 'Y'),
        capTaxProducts: (capReductionTaxFields['is_product'] === 'Y'),
      });
    
    case UPDATE_TAX_SETUP_INPUT: 
      return extend({}, state,
        {
          [action.payload.field]: action.payload.value
        }
      );
    default: 
      return state;
  }
};

export default taxSetup;
