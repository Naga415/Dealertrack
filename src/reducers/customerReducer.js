'use strict';
import { SET_CUSTOMER } from '../actions/actionTypes';
import ObjectUtil from '../utils/ObjectUtil';
import initialState from './initialState';

export default function customerReducer(customer = initialState.customer, action) {
  switch (action.type) {
    case SET_CUSTOMER:
      return ObjectUtil.clone(customer, action.payload.customer);
    default:
      return customer;
  }
}

