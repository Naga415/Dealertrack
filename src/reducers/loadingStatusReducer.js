'use strict';
import * as ActionTypes from '../actions/actionTypes';
import initialState from './initialState';

export default function loadingStatusReducer(state = initialState.loading, action) {
  switch (action.type) {
    case ActionTypes.INITIALIZATION_SUCCEEDED:
      return false;
    default:
      return state;
  }
}
