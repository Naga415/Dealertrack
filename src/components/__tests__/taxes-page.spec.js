import React from "react";
import renderer from "react-test-renderer";
import TaxesPage from "../taxes-page";
import extend from "lodash/extend";
import { initialState as taxSetupProps } from "../../reducers/tax-setup-reducers";

describe("TaxesPage", () => {
  it("renders <TaxesPage />", () => {
    const props = {
      htmlId: "TaxesPage",
      currentTab: 1,
      noTaxes: false,
      updateEditTaxProfileInput: () => {},
      updateCurrentTab: () => {},
      updateTaxSetupInput: () => {},
      actions: {},
      fetchTaxData: () => {},
      fetchFeeTaxData: () => {},
      taxSetup: extend({}, taxSetupProps)
    };

    const component = renderer.create(<TaxesPage {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
