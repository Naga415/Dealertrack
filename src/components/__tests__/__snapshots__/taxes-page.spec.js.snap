// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaxesPage renders <TaxesPage /> 1`] = `
<div
  id="edit-taxes-page"
>
  <div>
    <div
      className="row"
      id="edit-tax-profile"
    >
      <div
        className="col-xs-offset-half col-md-3 col-xs-4"
        id="noTaxes"
      >
        <div
          className="checkbox"
        >
          <label
            htmlFor="noTaxes"
          >
            <input
              checked={false}
              disabled={false}
              name="noTaxes"
              onChange={[Function]}
              type="checkbox"
            />
            No Taxes
          </label>
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxesProfile"
      >
        <div
          className="col-xs-12 form-group-taxes"
        >
          <label
            htmlFor="profile"
          >
            Profile
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-2--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-2--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-2"
        id="taxes-state-zipcode"
      >
        <label
          className="boldify zipcode-text"
          htmlFor={undefined}
        >
          ,
        </label>
      </div>
    </div>
  </div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="active"
    >
      <a
        onClick={[Function]}
      >
        Tax Setup 
      </a>
    </li>
    <li
      className={false}
    >
      <a
        onClick={[Function]}
      >
        Fee tax Setup 
      </a>
    </li>
    <li
      className={false}
    >
      <a
        onClick={[Function]}
      >
        Product category tax Setup
      </a>
    </li>
  </ul>
  <div
    className="tax-setup-container"
    id="edit-tax-setup"
  >
    <div
      className="row retail-row"
    >
      <div
        className="col-xs-3"
        id="taxes-retail-flat"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="retail-flat"
          >
            Retail Flat
          </label>
          <div
            className="input-group"
          >
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-usd"
              />
            </span>
            <input
              className="form-control "
              disabled={false}
              name="retail-flat"
              onChange={[Function]}
              type="text"
              value="0"
            />
          </div>
          
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxes-retail-trade-credit"
      >
        <div
          className="col-xs-12 form-group-taxes"
        >
          <label
            htmlFor="retail-trade-credit"
          >
            Retail-Trade Credit
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-3--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-3--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-sm-3 col-xs-4"
        id="taxes-retail-tarde-credit-amount"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="trade-credit-amountt"
          >
            Trade Credit Amount
          </label>
          <div
            className="input-group"
          >
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-usd"
              />
            </span>
            <input
              className="form-control "
              disabled={false}
              name="trade-credit-amountt"
              onChange={[Function]}
              type="text"
              value="0"
            />
          </div>
          
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-3"
        id="taxes-retail-tax-percent"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="retail-tax-percent"
          >
            Retail Tax
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              disabled={false}
              name="retail-tax-percent"
              onChange={[Function]}
              type="text"
              value="0"
            />
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-percent"
              />
            </span>
          </div>
          
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxes-retail-tax-method"
      >
        <div
          className="col-xs-10 form-group-taxes"
        >
          <label
            htmlFor="retail-tax-method"
          >
            Method
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-4--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-4--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-retail-tooltip"
        >
          <span>
            <div
              className="taxestooltip"
            >
              <i
                className="fa fa-info-circle"
              />
              <span
                className="tooltiptext"
              >
                Retail tax Method Options
              </span>
            </div>
          </span>
        </div>
      </div>
      <div
        className="col-xs-4"
      >
        Apply to
      </div>
      <div
        className="row lease-row"
      >
        <div
          className="col-xs-1"
          id="taxes-retail-tax-fees"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="retail-tax-fees"
            >
              <input
                checked={false}
                disabled={false}
                name="retail-tax-fees"
                onChange={[Function]}
                type="checkbox"
              />
              Fees
            </label>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-retail-tax-products"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="retail-tax-products"
            >
              <input
                checked={false}
                disabled={false}
                name="retail-tax-products"
                onChange={[Function]}
                type="checkbox"
              />
              Products
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-2"
        id="taxes-capitalize"
      >
        <div
          className="checkbox"
        >
          <label
            htmlFor="capitalize"
          >
            <input
              checked={false}
              disabled={false}
              name="capitalize"
              onChange={[Function]}
              type="checkbox"
            />
            Capitalize
          </label>
        </div>
      </div>
    </div>
    <hr />
    <div
      className="row"
    >
      <div
        className="col-xs-3"
        id="taxes-lease-flat"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="lease-flat"
          >
            Lease Flat
          </label>
          <div
            className="input-group"
          >
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-usd"
              />
            </span>
            <input
              className="form-control "
              disabled={false}
              name="lease-flat"
              onChange={[Function]}
              type="text"
              value="0"
            />
          </div>
          
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxes-lease-trade-credit"
      >
        <div
          className="col-xs-12 form-group-taxes"
        >
          <label
            htmlFor="lease-trade-credit"
          >
            Lease Trade Credit
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-5--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-5--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3 col-xs-3"
        id="taxes-lease-trade-credit-amount"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="trade-credit-amountt"
          >
            Trade Credit Amount
          </label>
          <div
            className="input-group"
          >
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-usd"
              />
            </span>
            <input
              className="form-control "
              disabled={false}
              name="trade-credit-amountt"
              onChange={[Function]}
              type="text"
              value="0"
            />
          </div>
          
        </div>
      </div>
    </div>
    <div
      className="row lease-row"
    >
      <div
        className="col-xs-3"
        id="taxes-lease-tax-percent"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="lease-tax-percent"
          >
            Lease Tax
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              disabled={false}
              name="lease-tax-percent"
              onChange={[Function]}
              type="text"
              value="0"
            />
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-percent"
              />
            </span>
          </div>
          
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxes-lease-tax-method"
      >
        <div
          className="col-xs-10 form-group-taxes"
        >
          <label
            htmlFor="lease-tax-method"
          >
            Method
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-6--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-6--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-lease-tooltip"
        >
          <span>
            <div
              className="taxestooltip"
            >
              <i
                className="fa fa-info-circle"
              />
              <span
                className="tooltiptext"
              >
                Lease tax Method Options
              </span>
            </div>
          </span>
        </div>
      </div>
      <div
        className="col-xs-4"
      >
        Apply to
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-1"
          id="taxes-lease-fees"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="lease-tax-fees"
            >
              <input
                checked={false}
                disabled={false}
                name="lease-tax-fees"
                onChange={[Function]}
                type="checkbox"
              />
              Fees
            </label>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-lease-products"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="lease-tax-products"
            >
              <input
                checked={false}
                disabled={false}
                name="lease-tax-products"
                onChange={[Function]}
                type="checkbox"
              />
              Products
            </label>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-3"
        id="taxes-mut-percent"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="monthly-use-tax-percent"
          >
            Monthly Use Tax
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              disabled={false}
              name="monthly-use-tax-percent"
              onChange={[Function]}
              type="text"
              value="0"
            />
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-percent"
              />
            </span>
          </div>
          
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxes-mut-method"
      >
        <div
          className="col-xs-10 form-group-taxes"
        >
          <label
            htmlFor="monthly-tax-method"
          >
            Method
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-7--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-7--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-mut-tooltip"
        >
          <span>
            <div
              className="taxestooltip"
            >
              <i
                className="fa fa-info-circle"
              />
              <span
                className="tooltiptext"
              >
                Monthly tax Method Options
              </span>
            </div>
          </span>
        </div>
      </div>
      <div
        className="col-xs-4"
      >
        Apply to
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-1"
          id="taxes-mut-fees"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="monthly-tax-fees"
            >
              <input
                checked={false}
                disabled={false}
                name="monthly-tax-fees"
                onChange={[Function]}
                type="checkbox"
              />
              Fees
            </label>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-mut-products"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="monthly-tax-products"
            >
              <input
                checked={false}
                disabled={false}
                name="monthly-tax-products"
                onChange={[Function]}
                type="checkbox"
              />
              Products
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-3"
        id="taxes-crt-percent"
      >
        <div
          className="form-group-taxes"
        >
          <label
            htmlFor="cap-reduction-tax-percent"
          >
            Cap Reduction Tax
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control "
              disabled={false}
              name="cap-reduction-tax-percent"
              onChange={[Function]}
              type="text"
              value="0"
            />
            <span
              className="input-group-addon"
            >
              <i
                className="fa fa-percent"
              />
            </span>
          </div>
          
        </div>
      </div>
      <div
        className="col-xs-5"
        id="taxes-crt-method"
      >
        <div
          className="col-xs-10 form-group-taxes"
        >
          <label
            htmlFor="cap-reduction-tax-method"
          >
            Method
          </label>
          <div
            className="Select Select--single is-clearable is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-8--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <style
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "input#undefined::-ms-clear {display: none;}",
                      }
                    }
                  />
                  <input
                    aria-activedescendant="react-select-8--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-crt-tooltip"
        >
          <span>
            <div
              className="taxestooltip"
            >
              <i
                className="fa fa-info-circle"
              />
              <span
                className="tooltiptext"
              >
                Cap Reduction tax Method Options
              </span>
            </div>
          </span>
        </div>
      </div>
      <div
        className="col-xs-4"
      >
        Apply to
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-1"
          id="taxes-crt-fees"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="cap-tax-fees"
            >
              <input
                checked={false}
                disabled={false}
                name="cap-tax-fees"
                onChange={[Function]}
                type="checkbox"
              />
              Fees
            </label>
          </div>
        </div>
        <div
          className="col-xs-1"
          id="taxes-crt-products"
        >
          <div
            className="checkbox"
          >
            <label
              htmlFor="cap-tax-products"
            >
              <input
                checked={false}
                disabled={false}
                name="cap-tax-products"
                onChange={[Function]}
                type="checkbox"
              />
              Products
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-3"
      >
        <label
          className="boldify"
          htmlFor={undefined}
        >
          Total: $ 0
        </label>
      </div>
      <div
        className="col-md-6 col-xs-6"
      >
        <label
          className="boldify col-xs-10 col-xs-offset-10 sales"
          htmlFor={undefined}
        >
          Sales $ 0 | MUT $ 0 |CRT $ 0
        </label>
      </div>
    </div>
  </div>
</div>
`;
